src/DynamoCore/Logging/AnalyticsService.cs

```diff
@@ -17,9 +17,24 @@ class AnalyticsService
         /// the Analytics service and application life cycle start is tracked.
         /// </summary>
         /// <param name="model">DynamoModel</param>
-        internal static void Start(DynamoModel model)
+        /// <param name="disableADPForProcessLifetime">Pass true to disable ADP for the lifetime of the Dynamo or host process</param>
+        internal static void Start(DynamoModel model, bool disableADPForProcessLifetime)
         {
             var client = new DynamoAnalyticsClient(model);
+            if (disableADPForProcessLifetime)
+            {
+                //if this returns false something has gone wrong.
+                //the client requested ADP be disabled, but we cannot disable it.
+                if (!ADPProcessSession.DisableADPForProcessLifetime())
```

Comment:
build will fail because this API doesn't exist until we merge the Analytics net wrapper PR